src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,45 @@
+package mate.academy.service.impl;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegisterException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+    
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> userFromDbOptional = userService.findByEmail(email);
+        if (userFromDbOptional.isPresent() && checkPassword(password, userFromDbOptional)) {
+            return userFromDbOptional.get();
+        }

```

Comment:
too complicated

you can instead:
```
User user = userService.findByEmail(email).orElseThrow("failed to find a user by email");
checkPassword(password, user);
return user;
```

and in `checkPassword()` you don't need to return anything - you just need to verify that the input password is equal to user's password - otherwise throw `AuthenticationException`